src/ir_Gree.cpp

```diff
@@ -190,49 +186,48 @@ void IRGreeAC::setModel(const gree_ac_remote_model_t model) {
   switch (model) {
     case gree_ac_remote_model_t::YAW1F:
     case gree_ac_remote_model_t::YBOFB: _model = model; break;
-    default: setModel(gree_ac_remote_model_t::YAW1F);
+    default: _model = gree_ac_remote_model_t::YAW1F;
   }
 }
 
 /// Get/Detect the model of the A/C.
 /// @return The enum of the compatible model.
-gree_ac_remote_model_t IRGreeAC::getModel(void) { return _model; }
+gree_ac_remote_model_t IRGreeAC::getModel() const { return _model; }
 
 /// Change the power setting to On.
-void IRGreeAC::on(void) { setPower(true); }
+void IRGreeAC::on() { setPower(true); }
 
 /// Change the power setting to Off.
-void IRGreeAC::off(void) { setPower(false); }
+void IRGreeAC::off() { setPower(false); }
 
 /// Change the power setting.
 /// @param[in] on true, the setting is on. false, the setting is off.
 /// @see https://github.com/crankyoldgit/IRremoteESP8266/issues/814
 void IRGreeAC::setPower(const bool on) {
-  setBit(&remote_state[0], kGreePower1Offset, on);
+  _.Power = on;
   // May not be needed. See #814
-  setBit(&remote_state[2], kGreePower2Offset,
-         on && _model != gree_ac_remote_model_t::YBOFB);
+  _.ModelA = on && _model == gree_ac_remote_model_t::YAW1F;
```

Comment:
Just for improved readability, can you add parentheses please?
e.g.
`_.ModelA = (on && _model == gree_ac_remote_model_t::YAW1F);`
It took me a second re-read of if to understand/make sure it was doing what I thought it was doing.